x-pack/plugins/enterprise_search/public/applications/app_search/components/curations/views/curation_suggestion/curation_suggestion_logic.ts

```diff
@@ -68,64 +74,92 @@ export const CurationSuggestionLogic = kea<
-          }
-        );
+        const suggestion = await getSuggestions(http, engineName, props.query);
+        const promotedIds: string[] = suggestion.promoted;
+        const documentDetailsResopnse = getDocumentDetails(http, engineName, promotedIds);
 
-        const suggestion = response.results[0];
+        let promises = [documentDetailsResopnse];
+        if (suggestion.curation_id) {
+          promises = [...promises, getCuration(http, engineName, suggestion.curation_id)];
+        }
 
-        const searchResponse = await http.post(
-          `/internal/app_search/engines/${engineName}/search`,
-          {
-            query: { query: '' },
-            body: JSON.stringify({
-              page: {
-                size: 100,
-              },
-              filters: {
-                id: suggestion.promoted,
-              },
-            }),
-          }
-        );
+        const [documentDetails, curation] = await Promise.all(promises);
 
         // Filter out docs that were not found and maintain promoted order
-        const promotedIds: string[] = suggestion.promoted;
-        const documentDetails = searchResponse.results;
         const suggestedPromotedDocuments = promotedIds.reduce((acc: Result[], id: string) => {
-          const found = documentDetails.find(
+          const found = documentDetails.results.find(
             (documentDetail: Result) => documentDetail.id.raw === id
           );
           if (!found) return acc;
           return [...acc, found];
         }, []);
 
         actions.onSuggestionLoaded({
-          suggestion: suggestion as CurationSuggestion,
+          suggestion,
           suggestedPromotedDocuments,
+          curation: curation || null,
         });
       } catch (e) {
         flashAPIErrors(e);
       }
     },
   }),
 });
+
+const getSuggestions = async (
+  http: HttpSetup,
+  engineName: string,
+  query: string
+): Promise<CurationSuggestion> => {
+  const response = await http.post(
+    `/internal/app_search/engines/${engineName}/search_relevance_suggestions/${query}`,
+    {
+      body: JSON.stringify({
+        page: {
+          current: 1,
+          size: 1,
+        },
+        filters: {
+          status: ['pending'],
+          type: 'curation',
+        },
+      }),
+    }
+  );
+
+  const suggestion = response.results[0] as CurationSuggestion;
+  return suggestion;
+};
+
+const getDocumentDetails = async (http: HttpSetup, engineName: string, documentIds: string[]) => {
+  return http.post(`/internal/app_search/engines/${engineName}/search`, {
+    query: { query: '' },
+    body: JSON.stringify({
+      page: {
+        size: 100,
+      },
+      filters: {
+        id: documentIds,
+      },
+    }),
+  });
+};
+
+const getCuration = async (http: HttpSetup, engineName: string, curationId: string) => {
+  return http.get(`/internal/app_search/engines/${engineName}/curations/${curationId}`, {
+    query: { skip_record_analytics: 'true' },
+  });
+};
```

Comment:
Worth testing these?

Reply:
Nah, there's no logic to them, I just pulled them out for readability.